packages/sdk/src/encryption/PublisherKeyExchange.ts

```diff
@@ -67,10 +75,13 @@ export class PublisherKeyExchange {
         })
         eventEmitter.on('messagePublished', (msg) => {
             if (msg.signatureType === SignatureType.ERC_1271) {
-                const address = msg.getPublisherId()
-                if (!this.erc1271ContractAddresses.has(address)) {
-                    logger.debug('Add ERC-1271 publisher', { address })
-                    this.erc1271ContractAddresses.add(address)
+                const signerUserId = msg.getPublisherId()
+                if (isEthereumAddressUserId(signerUserId)) {
```

Comment:
Maybe worth a warning log if we receive an ERC-2171 signature message but the publisher is not a valid ethereum address?

Reply:
This is a message published by this client. We ensure the validity at `MessageFactory:142`.